pkg/handlers/openapi/generate.go

```diff
@@ -0,0 +1,102 @@
+package openapi
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"net/url"
+	"os"
+	"strings"
+
+	"github.com/aserto-dev/aserto/pkg/cc"
+	"github.com/getkin/kin-openapi/openapi3"
+)
+
+// Test with: https://github.com/splunk/splunk-cloud-sdk-go/tree/master/services/action
+
+type GenerateOpenAPI struct {
+	Path string `arg:"" required:"" help:"path to openapi.yaml"`
+	Name string `arg:"" required:"" help:"path to openapi.yaml"`
+}
+
+func parseURI(uri string) []string {
+	result := []string{}
+	parts := strings.Split(uri, "/")
+	for _, part := range parts[1:] {
+		if strings.Contains(part, "{") {
+			var clean = strings.Replace(strings.Replace(part, "{", "", -1), "}", "", -1)
+			result = append(result, "__"+clean)
+		} else {
+			result = append(result, part)
+		}
+	}
+	return result
+}
+
+func generatePackageName(root, verb, uri string) string {
```

Comment:
I don't think there's any reason to perpetuate this naming pattern with a "root" prefix in every policy. It originated when we the authorizer loaded all policies into a single runtime and there had to be a way to disambiguate them. We no longer do that and I don't see the benefit of cluttering policy modules like that.

I think it should be optional and default to no having a root.

Reply:
Roots are still useful generally speaking to orient the user around the policy that they are using. Moreover, they are still in all the middleware libraries we have, so until we remove them everywhere I'm going to keep the root (verified this was reasonable with @gertd).